lib/wda/webdriveragent.js

```diff
@@ -283,12 +277,29 @@ class WebDriverAgent {
     this.noSessionProxy = new NoSessionProxy(proxyOpts);
   }
 
+  async ensureConnection () {
+    if (!this.realDevice || this.webDriverAgentUrl || this.iproxy) {
+      return;
+    }
+    if (isLocalHost(this.wdaBaseUrl)) {
+      this.iproxy = await this.createIProxy();
```

Comment:
the next step might be to add better error handling to `createIProxy`, so we can suggest which process to kill in order to free up the port

Reply:
The next step is creating a Factory class to handle the cached WDA instances more properly